src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
```

Comment:
You don't need to create methods addFirst and addLast with duplication of code. 
For implement logic of LinkedList, you can create private methods like indexValidation, unlink, findNodeByIndex, findIndexByValue. 
Try to refactor the code according to my comment and make it more readable and simplified